File: backend/app/schemas/document.py
"""
Document Pydantic Schemas
"""

from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional
from uuid import UUID


class DocumentStatus(BaseModel):
    """Document status enum"""

    UPLOADED = "uploaded"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"


class DocumentCreate(BaseModel):
    """Schema for creating a document"""

    file_name: str
    file_url: str
    mime_type: str
    file_size: int


class DocumentResponse(BaseModel):
    """Schema for document response"""

    id: UUID
    file_name: str
    file_url: str
    mime_type: str
    file_size: int
    status: str
    error_message: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class DocumentListResponse(BaseModel):
    """Schema for list of documents"""

    documents: list[DocumentResponse]
    total: int

File: frontend/src/types/document.ts
export interface Document {
  id: string
  file_name: string
  file_url: string
  mime_type: string
  file_size: number
  status: 'uploaded' | 'processing' | 'completed' | 'failed'
  error_message?: string
  created_at: string
  updated_at: string
}

export interface AIResult {
  id: string
  document_id: string
  doc_type: string
  confidence: number
  summary?: string
  risks?: any[]
  extracted_data: Record<string, any>
  ai_model?: string
  processing_time?: number
  created_at: string
}

export interface ChatMessage {
  question: string
  answer: string
}
